bin/j

```diff
@@ -0,0 +1,276 @@
+#!/usr/bin/env bash
+# Jira but as a quick and dirty CLI
+#
+# This script expects you to have a ~/.jira_credentials file with the following
+# contents:
+# ```yaml
+# user: "<your canonical email address>"
+# token: "<a valid Jira API token>"
+# board_id: <integer board id>
+# sprint_prefix: "<string prefix for you Jira pulse names>"  # e.g. "(Apps)"
+# ```
+#
+# See here for details of how to obtain a Jira API token:
+#   https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/
+#
+# If you are adding anything to this then see here for API docs:
+#   https://developer.atlassian.com/cloud/jira/software/rest/intro/#introduction
+#   https://docs.atlassian.com/software/jira/docs/api/REST/9.17.0/#api/2/
+USER="$(grep user ~/.jira_credentials | cut -d' ' -f2 | tr -d '"')"
+TOKEN="$(grep token ~/.jira_credentials | cut -d' ' -f2 | tr -d '"')"
+PREFIX="$(grep sprint_prefix ~/.jira_credentials | cut -d' ' -f2 | tr -d '"')"
+BOARD_ID="$(grep board_id ~/.jira_credentials | cut -d' ' -f2)"
+BACKLOG="UDENG"
+
+declare -A C=(
+  [red]='\033[31m'   [green]='\033[32m'  [yellow]='\033[33m'
+  [blue]='\033[34m'  [purple]='\033[35m' [cyan]='\033[36m'
+  [white]='\033[37m' [bold]='\033[1m'    [nc]='\033[0m'
+)
+C_HEADING="${C[cyan]}${C[bold]}"
+C_WARN="${C[yellow]}${C[bold]}"
+C_ERROR="${C[red]}${C[bold]}"
+
+
+# == helpers ==
+function cecho {
+  echo -e "${C[$1]}$2${C[nc]}"
+}
+function warn {
+  echo -e "${C_WARN}::${C[nc]} $*"
+}
+function heading {
+  echo -e "${C_HEADING}::${C[nc]} $*"
+}
+function warn_and_exit {
+  echo -e "${C_WARN}::${C[nc]} $*" && exit 0
+}
+function error_and_exit {
+  echo -e "${C_ERROR}error:${C[nc]} $*" && exit 1
+}
+
+function txt_from_editor {
+  local default_body=$1
+
+  fname=$(mktemp) || exit 1
+  echo "$default_body" >"$fname"
+  $EDITOR "$fname" </dev/tty >/dev/tty  # make sure that we have a TTY to work with for the editor
+  cat "$fname"
+  rm "$fname"
+}
+
+function require_external {
+  for prog in "$@"; do
+    [ -x "$(command -v "$prog")" ] || error_and_exit "'$prog' is required for this script to run"
+  done
+}
+
+
+# [ Actions ]
+
+function usage {
+  heading "j - quick and simple Jira"
+  cat <<EOF
+
+usage:
+  j current-sprint                 pretty print the details of the current sprint
+    summarise <tickets...>         pretty print ticket titles and status
+    show <ticket>                  show the details of a given ticket
+    new <type> <component> <title> open a new ticket using your \$EDITOR
+
+    [-h|--help]                    display this help message
+EOF
+}
+
+
+function create_issue_in_current_sprint {
+  local data description key sprint_id issue_type=$1 component=$2 summary=$3
```

Comment:
In fact that seems to not work with your recent refactor, if I add an `echo "type=$1 component=$2 summary=$3"` I get

```
$ ./j new Task Integration "Thunderbird 128.4.3esr update"
...
type=Task Integration Thunderbird 128.4.3esr update component= summary=
```